packages/flame/lib/src/experimental/geometry/shapes/polygon.dart

```diff
@@ -241,4 +243,35 @@ class Polygon extends Shape {
 
   @override
   String toString() => 'Polygon($vertices)';
+
+  @override
+  Vector2 randomPoint({Random? random, bool within = true}) {
+    final randomGenerator = random ?? randomFallback;
+    if (within) {
+      final result = Vector2.zero();
+      final min = aabb.min;
+      final max = aabb.max;
+
+      while (true) {
```

Comment:
I think there are ways to do this, at least for convex polygons
https://blogs.sas.com/content/iml/2020/10/21/random-points-in-polygon.html
but this is probably good enough, I wouldn't expect it to be slow for "normal" shapes

Reply:
Yeah, I wanted it to work for concave ones too, and there doesn't seem to be any good efficient algorithm for that.